### DISCUSSION TITLE
DHCP on wrong interface

### DISCUSSION THREAD

**Discussion by URZNI** [ltsp/ltsp#681] [Q&A]
Hello again,
I am currently trying to get LTSP to work on a dual NIC Setup with a /23 Subnet, because the number of clients is around 270.

eth0 = public server address (/24) in a internet-routed vlan 
eth1 = private server address/DHCP address in an isolated vlan

What i tried is to set a /23 IP (for example 10.0.1.254) on the interface eth1, specify the Server in the _/etc/ltsp/ltsp.conf_ ,
```
[Server]
SERVER="10.0.1.254"
```

set the DHCP-Range in the _/etc/dnsmasq.d/ltsp-dnsmasq.conf_ ,
`dhcp-range=eth1,10.0.0.1,10.0.1.200,12h`

and to make sure dnsmasq uses the correct interface, specified the interface in the _/etc/dnsmasq.conf_.
```
# If you want dnsmasq to listen for DHCP and DNS requests only on
# specified interfaces (and the loopback) give the name of the
# interface (eg eth0) here.
# Repeat the line for more than one interface.
interface=eth1
# Or you can specify which interface _not_ to listen on
except-interface=eth0
```

This did not work, DNSMASQ and LTSP gave the following messages:

DNSMASQ:
> no address range available for DHCP request via eth0

LTSP:
>  ltsp ltsp[1989]: Couldn't disable flow control for eth0
ltsp ltsp[1989]: Couldn't disable flow control for eth1

What is interesting is that even after removing the changes to the dnsmasq.conf and the ltsp-dnsmasq.conf, 
the messages were still appearing. I also had NAT set up manually via the ufw and also tried out NAT via the ltsp.conf but neither of them changed the outcome. Rebooting the machine did also not help.

My question is, does anybody know what I am doing wrong here
and what is the best way to set up a dual NIC Setup with a private /23 subnet, DHCP and NAT?

**Comment by alkisg** [comment]
Out of the box, LTSP doesn't support customizing the internal range or the subnet.
You will need to look at the source code and run similar commands manually for the subnet that you want.
You don't need to declare interfaces in dnsmasq.conf, it can deduce them from the IPs that you have in them.
You may ignore the flow control messages, they're not related to this issue.
